bundlewrap/items/__init__.py

```diff
@@ -532,7 +532,7 @@ def apply(
                 )):
                     self.fix(status_before)
             else:
-                if status_before.must_be_created:
+                if status_before.must_be_created and self.ITEM_TYPE_NAME != 'file':
```

Comment:
I think special-casing this is the way to go and preferable to creating some complicated abstraction for this case that no other item is going to use 👍🏻